### DISCUSSION TITLE
Coil usage in compose still isn't 100% clear

### DISCUSSION THREAD

**Discussion by ColtonIdle** [coil-kt/coil#811] [Q&A] [answered]
The example given shows

```
Image(
    painter = rememberImagePainter(
        data = "https://www.example.com/image.jpg",
        onExecute = ExecuteCallback { _, _ -> true },
        builder = {
            allowHardware(false)
            crossfade(true)
            placeholder(R.drawable.placeholder)
        }
    ),
    contentDescription = null,
    modifier = Modifier.size(128.dp)
)
```

What I don't understand is whether or not `allowHardware(false)` is something we should be using? I wasn't using this in view based coil? Also, The executeCallback is still unclear to me. I removed that section and everything still works.

So my question is. What's the baseline/standard/typical way to use coil-compose?

Is it just

```
Image(
    painter = rememberImagePainter(
        data = "https://www.example.com/image.jpg",
        builder = {
            crossfade(true)
        }
    ),
    contentDescription = null,
)
```

I do want to say that accompanist compose had a few issues in my app with images not loading intermittently, accompanist requiring a scale type for some reason to work properly, and required wayyy too much code to do a crossfade. I'm very happy to announce that all those issues went away when using coil-compose!

**Comment by colinrtwhite** [comment, marked as answer]
```kotlin
Image(
    painter = rememberImagePainter("https://www.example.com/image.jpg"),
    contentDescription = null,
    modifier = Modifier.size(128.dp)
)
```

is the simplest way to use Coil in Jetpack Compose. `builder` is only used to configure the `ImageRequest` object.

`allowHardware(false)` and `onExecute` are only shown in those examples for illustrative purposes. I removed the `allowHardware` call from the docs since it can be a bit confusing and is only used in specific cases.

**Reply by ColtonIdle** [reply]
Thanks @colinrtwhite I was hoping that `allowHardware` wasn't actually necessary in typical usage. Appreciate the help. Cheers
